artemis/src/main/java/tech/pegasys/artemis/state/BeaconState.java

```diff
@@ -437,23 +423,158 @@ public void exit_validator(BeaconState state, int index, int new_status) {
    * @param slot
    * @return
    */
-  private int get_beacon_proposer_index(BeaconState state, int slot) {
-    int[] first_committee = get_shard_committees_at_slot(state, slot).get(0).getCommittee();
-    return first_committee[slot % first_committee.length];
+  public static int get_beacon_proposer_index(BeaconState state, int slot) {
+    ArrayList<Integer> first_committee = get_shard_committees_at_slot(state, slot).get(0).getCommittee();
+    return first_committee.get(slot % first_committee.size());
   }
 
   /**
    * Returns the ``ShardCommittee`` for the ``slot``.
    * @param slot
    * @return
    */
-  private ArrayList<ShardCommittee> get_shard_committees_at_slot(BeaconState state, int slot) {
+  public static ArrayList<ShardCommittee> get_shard_committees_at_slot(BeaconState state, int slot) {
     int earliest_slot_in_array = toIntExact(state.getSlot()) - (toIntExact(state.getSlot()) % EPOCH_LENGTH)
         - EPOCH_LENGTH;
     assert earliest_slot_in_array <= slot;
     assert slot < (earliest_slot_in_array + EPOCH_LENGTH * 2);
 
-    return shard_committees_at_slots.get(slot - earliest_slot_in_array);
+    int index = slot - earliest_slot_in_array;
+    if (index < 0) {
+      index = state.shard_committees_at_slots.size() + index;
+    }
+    return state.shard_committees_at_slots.get(index);
+  }
```

Comment:
@akhila-raju I believe @dangerousfood was looking at this logic as well.  Did you two sync up?

Reply:
Yep that PR was closed; this addition is Java-specific.